tools/make_patch.c

```diff
@@ -0,0 +1,597 @@
+					// If we're at the end, we can get the symbol for the label
+					buffer[buffer_index] = '\0';
+					// Translate spaces to underscores
+					for (char *p = buffer; *p != '\0'; p++) {
+						if (*p == ' ') {
+							*p = '_';
+						}
+					}
+					// Look for the symbol starting with .VC_
+					char *searchlabel = xmalloc(strlen(".VC_") + strlen(buffer) + 1);
+					strcpy(searchlabel, ".VC_");
+					strcat(searchlabel, buffer);
+					current_hook = find_symbol(symbols, searchlabel);
+					if (!current_hook) {
+						fprintf(stderr, "Error: Cannot find symbol: %s\n", searchlabel);
+					}
+					free(searchlabel);
+					// Skip until the next newline
+					for (c = getc(file); c != EOF; c = getc(file)) {
+						putc(c, output);
+						if (c == '\n' || c == '\r') {
+							break;
+						}
+					}
+					buffer_index = 0;
+					break;
+				}
+				buffer[buffer_index++] = c;
+			}
+			break;
+
+		default:
+			putc(c, output);
+			line_pos++;
+		}
+	}
+
+	rewind(orig_rom);
+	rewind(new_rom);
+
+	fclose(file);
+	fclose(output);
+	free_buffer(buffer);
+	return patches;
+}
+
+int compare_patch(const void *patch1, const void *patch2) {
+	unsigned int patch1_offset = ((const struct Patch *)patch1)->offset;
+	unsigned int patch2_offset = ((const struct Patch *)patch2)->offset;
+	return patch1_offset > patch2_offset ? 1 : patch1_offset < patch2_offset ? -1 : 0;
+}
+
+bool verify_completeness(FILE *orig_rom, FILE *new_rom, struct Patches *patches) {
+	qsort(patches->data, patches->size, sizeof(*patches->data), compare_patch);
+	size_t index = 0;
+
+	for (size_t offset = 0; ; offset++) {
+		int orig_byte = getc(orig_rom);
+		int new_byte = getc(new_rom);
+		if (orig_byte == EOF || new_byte == EOF) {
+			return orig_byte == new_byte;
+		}
+		struct Patch *patch = patches->data + index;
+		if (index < patches->size && patch->offset == offset) {
+			if (fseek(orig_rom, patch->size, SEEK_CUR)) {
+				return false;
+			}
+			if (fseek(new_rom, patch->size, SEEK_CUR)) {
+				return false;
+			}
+			offset += patch->size;
+			index++;
+		} else if (orig_byte != new_byte) {
+			fprintf(stderr, "Value mismatch at decimal offset: %li\n", offset - 1);
+			fprintf(stderr, "Original ROM value: %x\n", orig_byte);
+			fprintf(stderr, "Patched ROM value: %x\n", new_byte);
+			fprintf(stderr, "Current patch start address: %x\n", patch->offset);
+			return false;
+		}
+	}
+}
+
+int main(int argc, char *argv[]) {
+	if (argc != 7) {
+		usage_exit(1);
+	}
+
+	struct Symbol *symbols = parse_symfile(argv[2]);
+	symbols = parse_constfile(argv[1], &symbols);
+
+	FILE *new_rom = xfopen(argv[3], 'r');
+	FILE *orig_rom = xfopen(argv[4], 'r');
+	struct Patches *patches = process_template(argv[5], new_rom, orig_rom, argv[6], symbols);
+
+	free_symbols(symbols);
+
+	if (!verify_completeness(orig_rom, new_rom, patches)) {
+		fprintf(stderr, "Warning: Not all ROM differences are defined by \"%s\"\n", argv[6]);
+	}
+
+	fclose(new_rom);
+	fclose(orig_rom);
+	free_patches(patches);
```

Comment:
```suggestion
	struct Patches *patches = process_template(argv[5], new_rom, orig_rom, argv[6], symbols);
	if (!verify_completeness(orig_rom, new_rom, patches)) {
		fprintf(stderr, "Warning: Not all ROM differences are defined by \"%s\"\n", argv[6]);
	}

	free_symbols(symbols);
	fclose(new_rom);
	fclose(orig_rom);
	free_patches(patches);
```

Reply:
Resolved in https://github.com/pret/pokecrystal/pull/882/commits/3ad8743ab680b1a72e0666fe0f6e5e14bda8ac17